lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+  attr_reader :name, :jokes
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << (joke)
+  end
+
+  def tell(friend, joke)
+    friend.learn(joke)
+  end
+
+  def joke_by_id(id)
```

Comment:
This is very much the right idea, but not quiiite the best choice of enumerable (although you certainly could have made it work).  `find_all` returns an array of the thing you're iterating over (joke instances, in this case), and you want a single joke.  `find` is the enumerable that returns a single instance that meets the criteria.  

Since you have an array with a joke inside, you can technically just do 
```
@jokes.find_all do |ident|
      id == ident.id
 end.first
```
which would give you the joke itself (not in an array).  

But, since this is exactly what the `find` method is for, we'd prefer that one instead: 
```
@jokes.find do |ident|
      id == ident.id
end 
```

Reply:
As written, this doesn't quite match the specifications of the interaction pattern-- you hacked your test for this method a little bit to make this work, which wouldn't be what we'd want to do.